src/sitemap/handler.js

```diff
@@ -0,0 +1,135 @@
+ * Checks the robots.txt file for a sitemap and returns the sitemap path if found.
+ *
+ * @async
+ * @param {string} protocol - The protocol (http or https) of the site.
+ * @param {string} domain - The domain of the site.
+ * @returns {Promise<{ path: string|null, reasons: string[] }>} - A Promise that resolves
+ * to an object containing the sitemap path and reasons for success or failure.
+ */
+export async function checkRobotsForSitemap(protocol, domain) {
+  const robotsUrl = `${protocol}://${domain}/robots.txt`;
+  try {
+    const robotsContent = await fetchContent(robotsUrl);
+    const sitemapMatch = robotsContent.match(/Sitemap:\s*(.*)/i);
+    return sitemapMatch && sitemapMatch[1]
+      ? { path: sitemapMatch[1].trim(), reasons: [] }
+      : { path: null, reasons: [ERROR_CODES.NO_SITEMAP_IN_ROBOTS] };
+  } catch (error) {
+    return { path: null, reasons: [ERROR_CODES.ROBOTS_NOT_FOUND] };
+  }
+}
+
+/**
+ * Checks the validity and existence of a sitemap by fetching its content.
+ *
+ * @async
+ * @param {string} sitemapUrl - The URL of the sitemap to check.
+ * @returns {Promise<Object>} - A Promise that resolves to an object
+ * representing the result of the sitemap check.
+ */
+export async function checkSitemap(sitemapUrl) {
+  try {
+    const sitemapContent = await fetchContent(sitemapUrl);
+    if (!sitemapContent) {
+      return {
+        existsAndIsValid: false,
+        reasons: [ERROR_CODES.SITEMAP_NOT_FOUND, ERROR_CODES.SITEMAP_EMPTY],
+      };
+    }
+    const isValidXml = sitemapContent.trim().startsWith('<?xml');
+    return {
+      existsAndIsValid: isValidXml,
+      reasons: isValidXml ? [] : [ERROR_CODES.SITEMAP_NOT_XML],
+    };
+  } catch (error) {
+    return { existsAndIsValid: false, reasons: [ERROR_CODES.FETCH_ERROR] };
+  }
+}
+
+/**
+ * Performs an audit for a specified site based on the audit request message.
+ *
+ * @async
+ * @param {Object} message - The audit request message containing the type, URL, and audit context.
+ * @param {Object} context - The context object containing configurations, services,
+ * and environment variables.
+ * @returns {Promise<Response>} - A Promise that resolves to a response object
+ * indicating the result of the audit process.
+ */
+
+export default async function audit(message, context) {
+  const { type, url: siteId, auditContext } = message;
+  const { dataAccess, log, sqs } = context;
+  const {
+    AUDIT_RESULTS_QUEUE_URL: queueUrl,
+  } = context.env;
+
+  try {
+    log.info(`Received ${type} audit request for siteId: ${siteId}`);
+
+    const site = await retrieveSiteBySiteId(dataAccess, siteId, log);
+    if (!site) {
+      log.error(`No site with siteId "${siteId}" exists.`);
+      return notFound('Site not found');
+    }
+
+    const baseURL = site.getBaseURL();
+    const auditResult = await findSitemap(baseURL);
+
+    log.info(`Audit result for ${baseURL}:\n${JSON.stringify(auditResult, null, 2)}`);
+
+    await sqs.sendMessage(queueUrl, {
```

Comment:
just a heads-up: we started saving the audit result before sending the sqs message. Example here: https://github.com/adobe/spacecat-audit-worker/blob/main/src/experimentation/handler.js#L77